pkg/pubsub/pubsub_init.go

```diff
@@ -0,0 +1,87 @@
+package pubsub
```

Comment:
can we add unit tests for init and system?

Reply:
Added some unit tests. I am adding some more tests still.

Reply:
@sozercan I think I added all the unit tests that make sense. Let me know if I missed any places.